atom/renderer/atom_sandboxed_renderer_client.cc

```diff
@@ -208,20 +211,27 @@ void AtomSandboxedRendererClient::DidCreateScriptContext(
   std::string left = "(function(binding, require) {\n";
   std::string right = "\n})";
   // Compile the wrapper and run it to get the function object
-  auto script = v8::Script::Compile(v8::String::Concat(
-      mate::ConvertToV8(isolate, left)->ToString(),
-      v8::String::Concat(node::preload_bundle_value.ToStringChecked(isolate),
-                         mate::ConvertToV8(isolate, right)->ToString())));
-  auto func =
-      v8::Handle<v8::Function>::Cast(script->Run(context).ToLocalChecked());
-  // Create and initialize the binding object
-  auto binding = v8::Object::New(isolate);
-  InitializeBindings(binding, context);
-  AddRenderBindings(isolate, binding);
-  v8::Local<v8::Value> args[] = {binding};
-  // Execute the function with proper arguments
-  ignore_result(
-      func->Call(context, v8::Null(isolate), node::arraysize(args), args));
+  auto maybe_script = v8::Script::Compile(
+      context, v8::String::Concat(
+                   mate::ConvertToV8(isolate, left)->ToString(),
+                   v8::String::Concat(
+                       node::preload_bundle_value.ToStringChecked(isolate),
+                       mate::ConvertToV8(isolate, right)->ToString())));
+  v8::Local<v8::Script> script;
+  v8::Local<v8::Value> result;
+  if (maybe_script.ToLocal(&script))
+    result = script->Run(context).ToLocalChecked();
```

Comment:
Perhaps this logic could be refactored to reuse the logic in `CreatePreloadScript`?